packages/angular/src/generators/directive/lib/validate-options.ts

```diff
@@ -0,0 +1,13 @@
+import type { Tree } from '@nrwl/devkit';
+import { getProjects } from '@nrwl/devkit';
+import { checkPathUnderProjectRoot } from '../../utils/path';
+import type { Schema } from '../schema';
+
+export function validateOptions(tree: Tree, options: Schema): void {
+  const projects = getProjects(tree);
+  if (!projects.has(options.project)) {
+    throw new Error(`Project "${options.project}" does not exist!`);
+  }
+
+  checkPathUnderProjectRoot(tree, options.project, options.path);
+}
```

Comment:
I wonder if we could make this shared between multiple generators. What are your thoughts?

Reply:
Possibly. As mentioned in the PR description: `Follow-up PRs will make this logic more consistent across component, directive and pipe generators`.

When making this consistent across the other similar generators, I'll extract whatever makes sense.